src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php

```diff
@@ -210,6 +236,56 @@ protected function registerMappingDrivers(array $objectManager, ContainerBuilder
         $container->setDefinition($this->getObjectManagerElementName($objectManager['name'].'_metadata_driver'), $chainDriverDef);
     }
 
+    /**
+     * @param class-string<MappingDriver> $driverClass
+     * @param string[]                    $dirs
+     *
+     * @return ?string service id, or null if not available
+     */
+    private function registerMappingClassLocatorService(string $driverClass, string $driverName, ContainerBuilder $container, array $dirs): ?string
+    {
+        $parameter = new \ReflectionParameter([$driverClass, '__construct'], 0);
+
+        $parameterType = TypeResolver::create()->resolve($parameter);
+
+        // It's possible that doctrine/persistence:^4.1 is installed with the older versions of ORM/ODM.
+        // In this case it's necessary to check for actual driver support.
+        if (!$parameterType->isIdentifiedBy(ClassLocator::class)) {
```

Comment:
What's the behavior when `ClassLocator` does not exist (persistence version < 4.1) but the driver constructor is updated? We need this condition to be true.

I see that `interface_exist(ClassLocator)` is already checked in `registerMappingDrivers`.